### PR TITLE
App Broker 2.0.x

### PR DESCRIPTION
**Pull request by spikymonkey**
App Broker 2.0.x

Major dependencies updates:
* Spring Boot 3
* Spring Framework 6
* Spring Cloud Open Service Broker 4.0.0-SNAPSHOT
* Java 17
* Gradle 7.6

Dockerfile:
* Update to Ubuntu Jammy
* Bump to OpenJDK 17
* Remove usage of deprecated `apt-key` tool
* Bump all other tools to latest versions (bosh, bbl, credhub etc.)

ATs:
* Pass required java buildpack env var to use Java 17 on CF
* Update `cf login` invocation to work with cf-cli v8

Small improvements:
* Reliable datetime comparison in `InMemoryServiceInstanceStateRepositoryTest`
* Update `.editorconfig` to match existing style
* Add `.sdkmanrc`

### CODE REVIEW

**Comment by spikymonkey** [comment]
This needs spring-cloud/spring-cloud-open-service-broker#367 to be merged before tests will pass.

**Review by royclarkson — CHANGES_REQUESTED** [review]
GitHub Actions matrix needs modification, otherwise LGTM!

File: .github/workflows/java-8-baseline.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  push:
+    branches: [ 1.6.x, 1.5.x, 1.4.x ]
+  pull_request:
+    branches: [ 1.6.x, 1.5.x, 1.4.x ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [ '8', '11' ]
```
Comment (royclarkson) [inline .github/workflows/java-8-baseline.yml:13]: We should run these on 17 as well

**Review by royclarkson — CHANGES_REQUESTED** [review]
Looks good, but have one question.

File: .github/workflows/java-8-baseline.yml (hunk 1 above)
Comment (spikymonkey) [reply .github/workflows/java-8-baseline.yml]: Done

**Review by royclarkson — APPROVED** [review]
Thanks!

**Pull request merged**